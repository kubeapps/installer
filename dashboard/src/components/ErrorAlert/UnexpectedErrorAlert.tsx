import * as React from "react";

import ErrorPageHeader from "./ErrorAlertHeader";

class UnexpectedErrorPage extends React.Component {
  public render() {
    return (
      <div className="alert alert-error margin-t-bigger">
        <ErrorPageHeader>Sorry! Something went wrong.</ErrorPageHeader>
        <div className="error__content margin-l-enormous">
          <p>Troubleshooting:</p>
          <ul className="error__troubleshooting">
            <li>Check for network issues.</li>
            <li>Check your browser's JavaScript console for errors.</li>
            <li>
              Check the health of Kubeapps components{" "}
              <code>kubectl get po --all-namespaces -l created-by=kubeapps</code>.
            </li>
            <li>
              <a href="https://github.com/kubeapps/kubeapps/issues/new" target="_blank">
                Open an issue on GitHub
              </a>{" "}
              if you think you've encountered a bug.
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

export default UnexpectedErrorPage;
