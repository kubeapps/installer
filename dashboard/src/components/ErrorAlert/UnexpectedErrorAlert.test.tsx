import { mount, shallow } from "enzyme";
import * as React from "react";

import UnexpectedErrorAlert from "./UnexpectedErrorAlert";

describe("when no text is passed", () => {
  it("renders a default troubleshooting info", () => {
    const wrapper = shallow(<UnexpectedErrorAlert />);
    expect(wrapper.text()).toContain("Troubleshooting");
    expect(wrapper.text()).toContain("Open an issue on GitHub");
    expect(wrapper).toMatchSnapshot();
  });
});

describe("when text is passed", () => {
  const defaultProps = {
    text: "This is my error message",
  };

  it("renders the message in a paragraph unless raw", () => {
    const wrapper = shallow(<UnexpectedErrorAlert {...defaultProps} />);
    const messageWrapper = wrapper.find("p");
    expect(messageWrapper).toExist();
    expect(messageWrapper.text()).toContain(defaultProps.text);
  });

  it("renders the message in a terminal if raw", () => {
    const wrapper = shallow(<UnexpectedErrorAlert {...defaultProps} raw={true} />);
    const messageWrapper = wrapper.find(".Terminal");
    expect(messageWrapper).toExist();
    expect(messageWrapper.text()).toContain(defaultProps.text);
  });

  it("uses the passed title or fallbacks to the default one", () => {
    let wrapper = mount(<UnexpectedErrorAlert {...defaultProps} />);
    expect(wrapper.text()).toContain("Sorry! Something went wrong");

    wrapper = mount(<UnexpectedErrorAlert {...defaultProps} title="My error" />);
    expect(wrapper.text()).not.toContain("Sorry! Something went wrong");
    expect(wrapper.text()).toContain("My error");
  });
});
