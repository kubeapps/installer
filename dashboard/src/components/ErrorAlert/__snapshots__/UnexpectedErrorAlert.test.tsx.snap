// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when no text is passed renders a default troubleshooting info 1`] = `
<div
  className="alert alert-error margin-t-bigger"
>
  <ErrorPageHeader
    icon={[Function]}
  >
    Sorry! Something went wrong.
  </ErrorPageHeader>
  <div
    className="error__content margin-l-enormous"
  >
    <div>
      <p>
        Troubleshooting:
      </p>
      <ul
        className="error__troubleshooting"
      >
        <li>
          Check for network issues.
        </li>
        <li>
          Check your browser's JavaScript console for errors.
        </li>
        <li>
          Check the health of Kubeapps components
           
          <code>
            kubectl get po --all-namespaces -l created-by=kubeapps
          </code>
          .
        </li>
        <li>
          <a
            href="https://github.com/kubeapps/kubeapps/issues/new"
            target="_blank"
          >
            Open an issue on GitHub
          </a>
           
          if you think you've encountered a bug.
        </li>
      </ul>
    </div>
  </div>
</div>
`;
